flink-runtime/src/main/java/org/apache/flink/runtime/leaderretrieval/ZooKeeperLeaderRetrievalDriverFactory.java

```diff
@@ -46,11 +51,13 @@ public ZooKeeperLeaderRetrievalDriverFactory(
     public ZooKeeperLeaderRetrievalDriver createLeaderRetrievalDriver(
             LeaderRetrievalEventHandler leaderEventHandler, FatalErrorHandler fatalErrorHandler)
             throws Exception {
+        driverReferenceCounter.incrementAndGet();
```

Comment:
I think each retrievalPath should have a seperate counter, do I understand it correctly?

Reply:
yes, because there's one `ZooKeeperLeaderRetrievalFactory` per path.